virttest/libvirt_xml/devices/tpm.py

```diff
@@ -84,6 +90,21 @@ def __init__(self, virsh_instance=base.base.virsh):
                                      subclass=self.ActivePCRBanks,
                                      subclass_dargs={
                                          'virsh_instance': virsh_instance})
+            accessors.XMLAttribute(property_name="source_type",
```

Comment:
Would you consider setting 'source' as XMLElementDict?

Reply:
Done. Thank you!